### PR TITLE
Added an option to pass connection params when initiating connection …

### PR DESCRIPTION
**Pull request by selvarajrajkanna**
…on strawberry channels test case

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to selvarajrajkanna for the PR 👏

Get it here 👉 https://beta.strawberry.rocks/release/(next)
```

**Comment by selvarajrajkanna** [comment]
Im following https://strawberry.rocks/docs/integrations/channels#testing to write test cases for my app. I notices that there is no parameter/method to pass connection_params when initiating the connection. 

In https://github.com/strawberry-graphql/strawberry/blob/main/strawberry/channels/testing.py file, on method gql_init() line 102, we are directly passing connection init message without any possibility to pass  connection_params.

**Comment by DoctorJohn** [comment]
Thanks for the PR @selvarajrajkanna . Would you be interested in adding a test case for this :)

Yup, we even have tests for the testing module. They can be found here: https://github.com/strawberry-graphql/strawberry/blob/main/tests/channels/test_testing.py

**Comment by selvarajrajkanna** [comment]
@DoctorJohn Added test case. Please review :)

**Review by DoctorJohn — APPROVED** [review]
Excellent, thanks for adding tests and the code optimization @selvarajrajkanna! I just noticed mentioning this in our docs would be great too. You're the expert on these changes so if you're interesting in adding a secting to the testing docs, feel free to do so. We'll merge this PR shortly. Let me know whether you would be interested in adding docs. That could either be done in one go with this PR or a separate one.

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Release type: minor
+
+Added support to pass connection_params as dictionary to GraphQLWebsocketCommunicator when testing [channels integration](https://strawberry.rocks/docs/integrations/channels#testing)
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
This release adds support to allow passing `connection_params` as dictionary to `GraphQLWebsocketCommunicator` class when testing [channels integration](https://strawberry.rocks/docs/integrations/channels#testing)
```

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+### sample code snippet
```
Comment (patrick91) [inline RELEASE.md:6]:
```suggestion
### Example
```

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+
+```
+GraphQLWebsocketCommunicator(
+    application=application,
+    path="/graphql",
+    connection_params={"username": "strawberry"}
+)
+```
```
Comment (patrick91) [inline RELEASE.md:15]:
````suggestion
```python
GraphQLWebsocketCommunicator(
    application=application,
    path="/graphql",
    connection_params={"username": "strawberry"}
)
```
````

**Review by patrick91 — APPROVED** [review]
Thanks! I left a couple of changes for the changelog 😊

File: RELEASE.md (hunk 3 above)
Comment (selvarajrajkanna) [reply RELEASE.md]: updated RELEASE.md file

File: RELEASE.md (hunk 1 above)
Comment (selvarajrajkanna) [reply RELEASE.md]: updated RELEASE.md file

**Comment by selvarajrajkanna** [comment]
@patrick91 updated the RELEASE.md file with recommendation :)

**Comment by selvarajrajkanna** [comment]
@DoctorJohn , Documentation isn't exactly my jam, but I'm a quick learner. Mind pointing me in the right direction?

**Comment by selvarajrajkanna** [comment]
@DoctorJohn , Essentially added your suggestion to the doc :)

**Comment by DoctorJohn** [comment]
> @DoctorJohn , Essentially added your suggestion to the doc :)

Awesome, thank you! I moved the new section in the docs a little further down so that the new subheading is scoped correctly. This is good to go now.

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**